README.md

```diff
@@ -4,37 +4,680 @@ Sylvia is the old name meaning Spirit of The Wood.
-and the [cw-multi-test](https://crates.io/crates/cw-multi-test) for testing them. Sylvia framework
-is built on top of them, so for creating contracts, you don't have to think about message structure,
-how their API is (de)serialized, or how to handle message dispatching. Instead, the API of your
-contract is a set of traits you implement on your SC type. The framework generates things like
-entry point structures, functions dispatching the messages, or even helpers for multitest. It
-allows for better control of interfaces, including validating their completeness in compile time.
+[CosmWasm](https://cosmwasm.com/) ecosystem core provides the base building
+blocks for smart contracts - the
+[cosmwasm-std](https://crates.io/crates/cosmwasm-std) for basic CW bindings, the
+[cw-storage-plus](https://crates.io/crates/cw-storage-plus) for easier state management,
+and the [cw-multi-test](https://crates.io/crates/cw-multi-test) for testing them.
+Sylvia framework is built on top of them, so for creating contracts, you don't
+have to think about message structure, how their API is (de)serialized, or how
+to handle message dispatching. Instead, the API of your contract is a set of
+traits you implement on your SC type. The framework generates things like entry
+point structures, functions dispatching the messages, or even helpers for multitest.
+It allows for better control of interfaces, including validating their completeness
+in compile time.
 
 Also, as a side effect, as Sylvia has all the knowledge about the contract API structure,
 it can generate many helpers - utilities for multitests or even queriers.
 
-## The state
+## Using in contracts
 
-Right now, Sylvia is ready to implement simple smart contracts. You can implement Instantiation,
-Execution, Query, and Migrate messages quickly. Sudo and reply messages are missing right now.
-Still, all of those can be covered by simply implementing their entry points and dispatching manually,
-so for basic functionality, Sylvia can always be used. Sylvia already supports overriding the
-chains' custom messages, so it is possible to use it with custom blockchains - except for handling custom
-queries and sudo messages. All messages can be generic, so creating a contract eligible to work
-across several different CosmWasm blockchains is also possible.
+Fist you need your contract crate, which should be a library crate:
 
-For now, only messages dispatching are generated - queriers utilities and MT helpers are on the TODO list, so expect them soon.
+```shell
+$ cargo new --lib ./my-crate
+     Created library `./my-crate` package
 
+```
+
+To use sylvia in the contract, you need to add couple dependencies - sylvia itself,
+and additionally: `cosmwasm-schema`, `schemars` and `cosmwasm_std`.
+
+```shell
+$ cargo add syvia cosmwasm-schema schemars cosmwasm-std
+...
+```
+
+You should also make sure your crate is compiling as `cdylib`, setting the proper
```

Comment:
This is useful instructions.  But two points:

1. I would break "create a new contract" into it's own markdown file (linked from this)
2. People shouldn't do all these steps, but could do something like [cw-template](https://github.com/CosmWasm/cw-template) to generate a basic sylvia contract (feel free to make it simpler / targetted to multi-contract repos)

Reply:
Splitting it into different files is definitely the way to go. However API is not stabilized yet, and it would be easier to check one file before release to update it, than the whole set of them.

Cargo template is a good idea. Just created #110. Will be part of 0.5 (I really don't want to add anything to 0.4).